app/models/manageiq/providers/amazon/storage_manager/ebs/refresh_parser_inventory_object.rb

```diff
@@ -59,6 +63,53 @@ def parse_snapshot(snap)
     }
   end
 
+  def link_volume_to_disk(volume_hash, attachments)
+    uid = volume_hash[:ems_ref]
+
+    attachments.each do |a|
+      if a['device'].blank?
+        _log.warn "#{log_header}: Volume: #{uid}, is missing a mountpoint, skipping the volume processing"
+        _log.warn "#{log_header}:   EMS: #{@ems.name}, Instance: #{a['instance_id']}"
+        next
+      end
+
+      dev = File.basename(a['device'])
+
+      vm = inventory_collections[:vms].find(a['instance_id']).data
+      unless vm
+        _log.warn "VM referenced by attachment (#{a['instance_id']} not found."
+        next
+      end
+
+      hardware = vm.hardware
```

Comment:
this would do query per each disk. You can rather load Hardware InventoryCollection, hardware uses the same id so then you can just do

hardware = inventory_collections[:hardwares].find(a['instance_id'])